### PR TITLE
Capybara fix n plus one

### PR DESCRIPTION
**Pull request by ahmidou09**
This pull request addresses the following tasks:

- Fixed the N+1 problem in fetching all posts and their comments for a user.
- Added Capybara integration tests for each view in the project.
- Implemented pagination on the User Post Index page.

### Changes Made
- Analyzed app logs before and after the N+1 problem fix. Screenshots are attached.
- Wrote integration tests using Capybara for user index, user show, user post index, and post show pages.
- Implemented pagination for the User Post Index page to handle a large number of posts.

### Screenshots
- Before:
before_1

- After:

after_1

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    visit users_path
+    expect(page).to have_css("img[src*='https://source.unsplash.com/featured/300x203']", count: 2)
+  end
+
+  it 'redirects to user show page when clicking on a user' do
+    visit users_path
+    click_link('John deo')
+    expect(current_path).to eq(user_path(User.last))
+  end
+end
```
Comment (mikethreels) [inline spec/features/user_index_spec.rb]: - [x] It seems you haven't included the below test in this file:
- Tests that assert you can see the number of posts each user has written.

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  it 'redirects to a users post show page when clicking on a users post' do
+    visit user_path(user1)
+    click_link 'Fifth post'
+    expect(current_path).to eq(user_post_path(user1, post5))
+  end
+
+  it 'redirects to the users post index page when clicking to see all posts' do
+    visit user_path(user1)
+    click_link 'See all posts'
+    expect(current_path).to eq(user_posts_path(user1))
```
Comment (mikethreels) [inline spec/features/user_show_spec.rb:59]: - [x] It seems you haven't included the below test in this file:
- Tests that asserts you can see the number of posts the user has written.

File: spec/features/post_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    expect(page).to have_content("Likes: #{post.likes_counter}")
+  end
+  it 'Display how many comments it has' do
+    visit user_post_path(user, post)
+    expect(page).to have_content("Comments: #{post.comments_counter}")
+  end
+  it 'Display the post body' do
+    visit user_post_path(user, post)
+    expect(page).to have_content(post.text)
+  end
```
Comment (mikethreels) [inline spec/features/post_show_spec.rb:34]: - [x] It seems you haven't included the below test in this file:
- Tests that assert you can see the username of each commentor.
- Tests that assert you can see the comment each commentor left.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Mehdi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job writing the tests

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @ahmidou09, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**